src/software/ai/evaluation/keep_away.h

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "shared/parameter/cpp_dynamic_parameters.h"
+#include "software/ai/passing/pass.h"
+#include "software/world/world.h"
+
+/**
+ * A measure of how good a point is for passing from, with specific emphasis on
+ * enemy proximity to the passer. Used for determining where to move to keep the
+ * ball away from enemy threats, while increasing the quality of a pass to the
+ * given receiver point in the pass.
+ *
+ * @param pass a pass to rate the receiver point quality for.
+ * @param enemy_team the enemy team
+ * @return a measure of quality of the passer point.
+ */
+double ratePasserPointForKeepAway(const Pass& pass, const Team& enemy_team);
+
+
+/**
+ * Find a point relatively close to the ball, that we can dribble it toward to maximize
+ * the quality of a pass to the receiver point in the best pass so far, while also
+ * trying to get far away from enemy robots.
```

Comment:
I'm wondering how well keep away transitions into passing. Can we have a test for that in shootorpassplaytest, for example?

Reply:
https://user-images.githubusercontent.com/41462969/121764265-8e6a5c00-caf7-11eb-8658-7f14585ccb44.mp4